poppy/tests/test_fresnel.py

```diff
@@ -533,3 +533,123 @@ def test_detector_in_fresnel_system(npix=256):
     assert psf[0].data.shape == (300, 300)
 
     assert psf[0].header['NAXIS1'] == 300
+
+def test_wavefront_conversions():
+    """ Test conversions between Wavefront and FresnelWavefront
+    in both directions.
+    """
+    import poppy
+
+    props = lambda wf: (wf.shape, wf.ispadded, wf.oversample, wf.pixelscale)
+
+    optic = poppy.CircularAperture()
+    w = poppy.Wavefront(diam=4*u.m)
+    w*= optic
+
+    fw = poppy.FresnelWavefront(beam_radius=2*u.m)
+    fw*= optic
+
+    # test convert from Fraunhofer to Fresnel
+    fw2 = poppy.FresnelWavefront.from_wavefront(w)
+    assert props(fw)==props(fw2)
+    #np.testing.assert_allclose(fw.wavefront, fw2.wavefront)
+
+    # test convert from Fresnel to Fraunhofer
+    w2 = poppy.Wavefront.from_fresnel_wavefront(fw)
+    assert props(w)==props(w2)
+
+
+def test_CompoundOpticalSystem_fresnel(npix=128, display=False):
+    """ Test that the CompoundOpticalSystem container works for Fresnel systems
+
+    Parameters
+    ----------
+    npix : int
+        Number of pixels for the pupil sampling. Kept small by default to
+        reduce test run time.
+    """
+
+    import poppy
+
+    opt1 = poppy.SquareAperture()
+    opt2 = poppy.CircularAperture(radius=0.55)
+
+    # a single optical system
+    osys = poppy.FresnelOpticalSystem(beam_ratio=0.25, npix=npix)
```

Comment:
beam ratio seems like it adds confusion for folks not coming from PROPER, should we start defaulting to oversample and just have an optional keyword for beam ratio, which throws an error if both are defined?

Reply:
I'm fine with that - and I agree having a more consistent API between the two versions would be beneficial. 
Right now the FresnelOpticalSystem class actually doesn't have an oversample keyword - it just has beam_ratio. I can't even remember why we did it that way since FresnelWavefront has oversample.  (In retrospect I also don't even like the keyword name 'oversample' in this sense, since 2 for Nyquist is well sampled rather than oversampled. But that too would break a lot of legacy code to change.)

That said, while I agree with this proposed change, I think it ought to be a separate PR from this one.